platforms/software/dependency-management/src/main/java/org/gradle/internal/component/resolution/failure/transform/SourceVariantData.java

```diff
@@ -40,4 +40,13 @@ public String getVariantName() {
     public ImmutableAttributes getAttributes() {
         return attributes;
     }
+
+    private String quoteVariantName(String variantName) {
+        int variantIdx = variantName.indexOf(" variant ");
+        if (variantIdx == -1) {
+            return variantName;
+        } else {
+            return variantName.substring(0, variantIdx + 9) + "'" + variantName.substring(variantIdx + 9) + "'";
+        }
```

Comment:
String manipulation like this seems really wrong to me. 

I don't think we should do this.

Reply:
We don't have to, but I like how it makes the output look.  It's more consistent to quote the variant names too.

Reply:
I'll separate this into a separate method, to make it obvious what's happening.